### PR TITLE
HIVE-26455: Remove PowerMock from hive-exec

### PR DESCRIPTION
**Pull request by InvisibleProgrammer**
Remove PowerMock from Hive-Exec, use mockito-inline and upgrade mockito version to 4.11.

### What changes were proposed in this pull request?
- PowerMock and mockito-core dependency is removed
- Mockito-inline dependency added
- Refactor replicaion task code to be able to unit test - added a replication logger factory
- System class mocking is removed - A time util class added to be able to do tests based on system time
- Task related tests modified a little bit: instead of checking the log entries in the given metric tests, it just check if the startLog and endLog methods are properly called.
- Static mocking in mockito-inline is safer than it was in PowerMock. It is scoped based and it used inside try blocks or the mocks are closed explicity.
- Test runners are replaced to MockitoJUnitRunner
- PreperaForTest, PrepareOnlyThisForTest and PowerMockIgnore annotations are removed - they are PowerMock only

And some changes because of mockito upgrade (as it deprecated/refactored significant part of it's API)
- Argument matchers moved to org.mockito.ArgumentMatchers
- any... matchers (like anyList) got rid of their parameters
- New class is introduced, ReflectionUtil - as FieldSetter is completely removed from mockito

### Why are the changes needed?
- PowerMock seems to be a dead project.
- Prerequisite for moving to Java 11

### Does this PR introduce _any_ user-facing change?
No

### Is the change a dependency upgrade?
Yes

### How was this patch tested?
Run the existing precommit tests.

### CODE REVIEW

File: ql/src/test/org/apache/hadoop/hive/ql/testutil/ReflectionUtil.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  /**
+   * Sets a declared field in a given object. It finds the field, even if it is declared in a super class.
+   * @param object target instance
+   * @param field name of the field to set
+   * @param value new value
+   * @throws RuntimeException in case the field is not found or cannot be set.
+   */
+  public static void setInAllFields(Object object, String field, Object value) {
+    try {
+      Field fieldToChange = Arrays.stream(FieldUtils.getAllFields(Cleaner.class))
```
Comment (InvisibleProgrammer) [inline ql/src/test/org/apache/hadoop/hive/ql/testutil/ReflectionUtil.java:61]: I have found some leftovers in this method. Let me fix it.

**Review by zratkai — CHANGES_REQUESTED** [review]

File: ql/src/java/org/apache/hadoop/hive/ql/util/TimeUtil.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package org.apache.hadoop.hive.ql.util;
+
+/**
+ * Makes it possible to mock test your code.
+ */
+public class TimeUtil {
+  public long getNanoTime(){
+    return System.nanoTime();
```
[the remark this answers is not in the document]
Comment (InvisibleProgrammer) [reply ql/src/java/org/apache/hadoop/hive/ql/util/TimeUtil.java:26]: The original code used System.nanoTime() directly. But we have a test case that depends on the time but mockito doesn't allow mocking System class at all. 

As far I know, the best pattern to mock static system calls is to create some wrapper class, use that wrapper class in our object that we want to test and replace it's behaviour in our tests. 

Do you have a better approach on that?

[the remark this answers is not in the document]
Comment (zratkai) [reply ql/src/java/org/apache/hadoop/hive/ql/util/TimeUtil.java:26]: Ah, ok, for this purpose I can understand.

**Review by zratkai — APPROVED** [review]

**Review by rkirtir — APPROVED** [review]

File: ql/src/java/org/apache/hadoop/hive/ql/util/TimeUtil.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.util;
+
+/**
+ * Makes it possible to mock test your code.
+ */
+public class TimeUtil {
```
Comment (ayushtkn) [inline ql/src/java/org/apache/hadoop/hive/ql/util/TimeUtil.java:24]: Why is this class required? For mocking? Can't we use the existing util class like ```Time.monotonicNowNanos()```? rather than introducing our own

File: ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestAtlasDumpTask.java [hunk 4]
```diff
@@ -92,48 +90,40 @@ public class TestAtlasDumpTask {
...
-    Mockito.verify(logger,
-        Mockito.times(2)).info(replStateCaptor.capture(),
-        eventCaptor.capture(), eventDetailsCaptor.capture());
-    Assert.assertEquals("REPL::{}: {}", replStateCaptor.getAllValues().get(0));
-    Assert.assertEquals("ATLAS_DUMP_START", eventCaptor.getAllValues().get(0));
-    Assert.assertEquals("ATLAS_DUMP_END", eventCaptor.getAllValues().get(1));
-    Assert.assertTrue(eventDetailsCaptor.getAllValues().get(1).toString(), eventDetailsCaptor.getAllValues().get(0)
-        .toString().contains("{\"dbName\":\"srcDB\",\"dumpStartTime"));
-    Assert.assertTrue(eventDetailsCaptor
-        .getAllValues().get(1).toString().contains("{\"dbName\":\"srcDB\",\"dumpEndTime\""));
```
Comment (ayushtkn) [inline ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestAtlasDumpTask.java:127]: Where are we validating these things now?

File: ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestAtlasLoadTask.java [hunk 5]
```diff
@@ -51,32 +55,21 @@ public class TestAtlasLoadTask {
...
-            Mockito.times(2)).info(replStateCaptor.capture(),
-            eventCaptor.capture(), eventDetailsCaptor.capture());
-    Assert.assertEquals("REPL::{}: {}", replStateCaptor.getAllValues().get(0));
-    Assert.assertEquals("ATLAS_LOAD_START", eventCaptor.getAllValues().get(0));
-    Assert.assertEquals("ATLAS_LOAD_END", eventCaptor.getAllValues().get(1));
-    Assert.assertTrue(eventDetailsCaptor.getAllValues().get(0)
-            .toString().contains("{\"sourceDbName\":\"srcDB\",\"targetDbName\":\"tgtDB\",\"loadStartTime\":"));
-    Assert.assertTrue(eventDetailsCaptor
-            .getAllValues().get(1).toString().contains("{\"sourceDbName\":\"srcDB\",\"targetDbName\""
-                    + ":\"tgtDB\",\"numOfEntities\":0,\"loadEndTime\""));
```
Comment (ayushtkn) [inline ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestAtlasLoadTask.java:80]: removing assertions?

File: ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestRangerDumpTask.java [hunk 6]
```diff
@@ -124,50 +127,39 @@ public void testSuccessNonEmptyRangerPolicies() throws Exception {
...
-    Mockito.verify(logger,
-        Mockito.times(2)).info(replStateCaptor.capture(),
-        eventCaptor.capture(), eventDetailsCaptor.capture());
-    Assert.assertEquals("REPL::{}: {}", replStateCaptor.getAllValues().get(0));
-    Assert.assertEquals("RANGER_DUMP_START", eventCaptor.getAllValues().get(0));
-    Assert.assertEquals("RANGER_DUMP_END", eventCaptor.getAllValues().get(1));
-    Assert.assertTrue(eventDetailsCaptor.getAllValues().get(0)
-        .toString().contains("{\"dbName\":\"testdb\",\"dumpStartTime"));
-    Assert.assertTrue(eventDetailsCaptor
-        .getAllValues().get(1).toString().contains("{\"dbName\":\"testdb\",\"actualNumPolicies\":0,\"dumpEndTime\""));
```
Comment (ayushtkn) [inline ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestRangerDumpTask.java:171]: removed?

File: ql/src/test/org/apache/hadoop/hive/ql/exec/util/TestRetryable.java [hunk 7]
```diff
@@ -340,7 +345,7 @@ public void testRetrySuccessSecureCallable() throws Throwable {
     ArgumentCaptor<PrivilegedExceptionAction> privilegedActionArgumentCaptor
       = ArgumentCaptor.forClass(PrivilegedExceptionAction.class);
     Mockito.verify(userGroupInformation,
-      Mockito.times(3)).doAs(privilegedActionArgumentCaptor.capture());
+      Mockito.times(1)).doAs(privilegedActionArgumentCaptor.capture());
```
Comment (ayushtkn) [inline ql/src/test/org/apache/hadoop/hive/ql/exec/util/TestRetryable.java:348]: why did the value change?

File: ql/src/test/org/apache/hadoop/hive/ql/parse/repl/TestCopyUtils.java [hunk 8]
```diff
@@ -97,151 +96,159 @@ public void shouldThrowExceptionOnDistcpFailure() throws Exception {
...
+            MockedStatic<Utils> utilsMockedStatic = mockStatic(Utils.class)
+      ) {
+      fileUtilsMockedStatic.when(() -> FileUtils.distCp(same(fs), anyList(), same(destination),
+                            anyBoolean(), eq(null), same(conf),
+                            same(ShimLoader.getHadoopShims())))
+          .thenReturn(false);
+      utilsMockedStatic.when(Utils::getUGI).thenReturn(mock(UserGroupInformation.class));
+
+      copyUtils.doCopy(destination, srcPaths);
+    }
```
Comment (ayushtkn) [inline ql/src/test/org/apache/hadoop/hive/ql/parse/repl/TestCopyUtils.java:113]: this doesn't look formatted properly
```
    when(source.getFileSystem(same(conf))).thenReturn(fs);
    try (MockedStatic<FileUtils> fileUtilsMockedStatic = mockStatic(FileUtils.class);
        MockedStatic<Utils> utilsMockedStatic = mockStatic(Utils.class)) {
      fileUtilsMockedStatic.when(
          () -> FileUtils.distCp(same(fs), anyList(), same(destination), anyBoolean(), eq(null), same(conf),
              same(ShimLoader.getHadoopShims()))).thenReturn(false);
      utilsMockedStatic.when(Utils::getUGI).thenReturn(mock(UserGroupInformation.class));

      copyUtils.doCopy(destination, srcPaths);
    }
```

File: ql/src/java/org/apache/hadoop/hive/ql/util/TimeUtil.java (hunk 3 above)
Comment (InvisibleProgrammer) [reply ql/src/java/org/apache/hadoop/hive/ql/util/TimeUtil.java:24]: Thx for the tip. I didn't know that we have this. As I see it is in apache-common and it is already a dependency of Hive.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/util/TestRetryable.java (hunk 7 above)
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/exec/util/TestRetryable.java:348]: TBH that is a behaviour that I haven't found an answer yet: My best guess is that there is some strange behaviour with the argument captors and powermock. I remember that case from last November: when I just simple changed the test runner from PowerMockRunner to MockitoJUnitRunner, it started to provide 1 as an answer. I debugged the test and found out 1 is the correct answer. It really runs it only once. So that I reverted the change, debugged it and what I found? When I ran it with PowerMock again. the execution stopped only once. And still, the times(3) went green. 
I really don't know what PowerMock does in that case. And only at that test.

File: ql/src/test/org/apache/hadoop/hive/ql/parse/repl/TestCopyUtils.java (hunk 8 above)
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/parse/repl/TestCopyUtils.java]: Replaced it. TBH, Now I think it is harder to read the code and interpret what it exactly does. But I know you know that code base way better than I am so I assume you have confidence about our formatting patterns as well. 
Do you want me to check and change all the mocking occurrences in the PR? 
As I understood your recommendation there are two issues: 
- try block should be more compact
- shouldn't put .thenReturn clauses into new line at methods with long parameter list.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestAtlasDumpTask.java (hunk 4 above)
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestAtlasDumpTask.java:127]: Tests are refactored based on TestReplicationScenarios, by using log appenders.

File: ql/src/test/org/apache/hadoop/hive/ql/testutil/ReflectionUtil.java [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+import java.lang.reflect.Field;
+import java.util.Arrays;
+
+import static java.lang.String.format;
+
+/**
+ * Various helper methods for mocking.
+ * <b>Please use it only in tests, not in production code!</b>
+ */
+public class ReflectionUtil {
```
Comment (SourabhBadhya) [inline ql/src/test/org/apache/hadoop/hive/ql/testutil/ReflectionUtil.java:32]: There is already a ReflectionUtil here - 
https://github.com/apache/hive/blob/master/common/src/java/org/apache/hive/common/util/ReflectionUtil.java

I think we must extend that rather than creating a new class.

File: ql/pom.xml [hunk 10]
```diff
@@ -809,20 +808,8 @@
...
-      <artifactId>powermock-module-junit4</artifactId>
-      <version>${powermock.version}</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.powermock</groupId>
-      <artifactId>powermock-api-mockito2</artifactId>
-      <version>${powermock.version}</version>
+      <artifactId>mockito-inline</artifactId>
+      <version>4.11.0</version>
```
[the remark this answers is not in the document]
Comment (InvisibleProgrammer) [reply ql/pom.xml:812]: It is planned to do one of these after all the module changes are merged: 
- bring back the mockito dependency to the root pom
- OR just set the version property in the root pom. 

I want to discuss it after all modules are merged because there are pros and cons about moving back it to the root pom.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/repl/ranger/TestRangerRestClient.java [hunk 11]
```diff
@@ -54,23 +55,26 @@ public class TestRangerRestClient {
...
-    Mockito.when(UserGroupInformation.getLoginUser()).thenReturn(userGroupInformation);
-    Mockito.when(userGroupInformation.doAs((PrivilegedAction<Object>) Mockito.any())).thenCallRealMethod();
-    Mockito.when(userGroupInformation.doAs((PrivilegedExceptionAction<Object>) Mockito.any())).thenCallRealMethod();
+    userGroupInformationMockedStatic = mockStatic(UserGroupInformation.class);
+
+    userGroupInformationMockedStatic.when(UserGroupInformation::getLoginUser).thenReturn(userGroupInformation);
+
     Mockito.when(mockClient.getRangerExportUrl(Mockito.anyString(), Mockito.anyString(), Mockito.anyString()))
-      .thenCallRealMethod();
-    Mockito.when(mockClient.getRangerImportUrl(Mockito.anyString(), Mockito.anyString()))
```
[the remark this answers is not in the document]
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/exec/repl/ranger/TestRangerRestClient.java:65]: No. It was a leftover. Mockito is way more restrictive than PowerMock and it doesn't allow stubbing. If I put it back into the code, it won't even run the tests, it gets something like UnnecessaryStubbingException.

File: ql/src/test/org/apache/hadoop/hive/ql/testutil/ReflectionUtil.java (hunk 9 above)
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/testutil/ReflectionUtil.java:32]: As I see, the only common thing in the ReflectionUtil class in the common module and this, is the name of the class. Anyway, it can be moved. But again, based on their code, they have way different purpose.

Comment (SourabhBadhya) [reply ql/src/test/org/apache/hadoop/hive/ql/testutil/ReflectionUtil.java:32]: I think its best to add in the existing file since other modules can also use it from `hive-common` rather than putting this in  `hive-exec`.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/repl/ranger/TestRangerRestClient.java [hunk 12]
```diff
@@ -54,23 +55,26 @@ public class TestRangerRestClient {
...
-      .thenCallRealMethod();
+            .thenCallRealMethod();
     Mockito.when(conf.getTimeVar(HiveConf.ConfVars.REPL_RETRY_INTIAL_DELAY, TimeUnit.SECONDS)).thenReturn(1L);
     Mockito.when(conf.getTimeVar(HiveConf.ConfVars.REPL_RETRY_TOTAL_DURATION, TimeUnit.SECONDS)).thenReturn(20L);
     Mockito.when(conf.getTimeVar(HiveConf.ConfVars.REPL_RETRY_JITTER, TimeUnit.SECONDS)).thenReturn(1L);
     Mockito.when(conf.getTimeVar(HiveConf.ConfVars.REPL_RETRY_MAX_DELAY_BETWEEN_RETRIES, TimeUnit.SECONDS))
-      .thenReturn(10L);
-    Mockito.when(conf.getFloat(HiveConf.ConfVars.REPL_RETRY_BACKOFF_COEFFICIENT.varname, 1.0f))
```
[the remark this answers is not in the document]
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/exec/repl/ranger/TestRangerRestClient.java:72]: No. It was a leftover. Mockito is way more restrictive than PowerMock and it doesn't allow stubbing. If I put it back into the code, it won't even run the tests, it gets something like UnnecessaryStubbingException.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestAtlasLoadTask.java [hunk 13]
```diff
@@ -55,28 +56,24 @@ public void testAtlasLoadMetrics() throws Exception {
...
     Mockito.when(conf.getBoolVar(HiveConf.ConfVars.HIVE_IN_TEST_REPL)).thenReturn(true);
-    Logger logger = Mockito.mock(Logger.class);
-    Whitebox.setInternalState(ReplState.class, logger);
+    Logger logger = LoggerFactory.getLogger("ReplState");
+    StringAppender appender = StringAppender.createStringAppender(null);
+    appender.addToLogger(logger.getName(), Level.INFO);
+    appender.start();
     AtlasReplInfo atlasReplInfo = new AtlasReplInfo("http://localhost:21000/atlas", "srcDB",
-        "tgtDB", "srcCluster", "tgtCluster", new Path("hdfs://tmp"), null, conf);
+            "tgtDB", "srcCluster", "tgtCluster", new Path("hdfs://tmp"), null, conf);
```
Comment (SourabhBadhya) [inline ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestAtlasLoadTask.java:64]: Unnecessary change.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestRangerDumpTask.java [hunk 14]
```diff
@@ -124,50 +123,45 @@ public void testSuccessNonEmptyRangerPolicies() throws Exception {
...
+    StringAppender appender = StringAppender.createStringAppender(null);
+    appender.addToLogger(logger.getName(), Level.INFO);
+    appender.start();
     RangerExportPolicyList rangerPolicyList = new RangerExportPolicyList();
     rangerPolicyList.setPolicies(new ArrayList<RangerPolicy>());
     Mockito.when(mockClient.exportRangerPolicies(Mockito.anyString(), Mockito.anyString(), Mockito.anyString(),
-      Mockito.any()))
-      .thenReturn(rangerPolicyList);
+                    Mockito.any()))
+            .thenReturn(rangerPolicyList);
```
Comment (SourabhBadhya) [inline ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestRangerDumpTask.java:152]: Unnecessary change.

File: ql/src/test/org/apache/hadoop/hive/ql/parse/repl/load/message/TestPrimaryToReplicaResourceFunction.java [hunk 15]
```diff
@@ -76,30 +78,37 @@ public void setup() {
...
-        "hdfs://localhost:9000/user/someplace/ab.jar#e094828883"));
-
+            "hdfs://localhost:9000/user/someplace/ab.jar#e094828883"));
     assertThat(resourceUri.getUri(),
-        is(equalTo(
-            "hdfs://somehost:9000/someBasePath/withADir/replicadbname/somefunctionname/" + String
-                .valueOf(0L) + "/ab.jar")));
+            is(equalTo(
+                    "hdfs://somehost:9000/someBasePath/withADir/replicadbname/somefunctionname/" + String
+                            .valueOf(0L) + "/ab.jar")));
```
Comment (SourabhBadhya) [inline ql/src/test/org/apache/hadoop/hive/ql/parse/repl/load/message/TestPrimaryToReplicaResourceFunction.java:112]: Unnecessary changes.

**Review by SourabhBadhya — COMMENTED** [review]
LGTM but there are some unnecessary changes. Please rectify it.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestAtlasDumpTask.java [hunk 16]
```diff
@@ -223,36 +222,39 @@ public void testAtlasServerEntityRetryExhausted() throws AtlasServiceException {
...
-    PowerMockito.verifyStatic(ConfigurationConverter.class, Mockito.times(1));
-    ConfigurationConverter.getConfiguration(propsCaptor.capture());
-    Assert.assertEquals("20", propsCaptor.getValue().getProperty(
-            AtlasRestClientBuilder.ATLAS_PROPERTY_CONNECT_TIMEOUT_IN_MS));
-    Assert.assertEquals("500", propsCaptor.getValue().getProperty(
-            AtlasRestClientBuilder.ATLAS_PROPERTY_READ_TIMEOUT_IN_MS));
+    try (
+            MockedStatic<UserGroupInformation> userGroupInformationMockedStatic = mockStatic(UserGroupInformation.class);
+            MockedStatic<ConfigurationConverter> configurationConverterMockedStatic = mockStatic(ConfigurationConverter.class)
+            ) {
```
[the remark this answers is not in the document]
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestAtlasDumpTask.java]: Modified.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestRangerLoadTask.java [hunk 17]
```diff
@@ -111,32 +113,34 @@ public void testSuccessNonEmptyRangerPolicies() throws Exception {
...
+            + "\"Exported by\":\"hive\",\"Export time\":\"May 5, 2020, 8:55:03 AM\",\"Ranger apache version\""
+            + ":\"2.0.0.7.2.0.0-61\"},\"policies\":[{\"service\":\"cm_hive\",\"name\":\"db-level\",\"policyType\":0,"
+            + "\"description\":\"\",\"isAuditEnabled\":true,\"resources\":{\"database\":{\"values\":[\"aa\"],"
+            + "\"isExcludes\":false,\"isRecursive\":false},\"column\":{\"values\":[\"id\"],\"isExcludes\":false,"
+            + "\"isRecursive\":false},\"table\":{\"values\":[\"*\"],\"isExcludes\":false,\"isRecursive\":false}},"
+            + "\"policyItems\":[{\"accesses\":[{\"type\":\"select\",\"isAllowed\":true},{\"type\":\"update\","
+            + "\"isAllowed\":true}],\"users\":[\"admin\"],\"groups\":[\"public\"],\"conditions\":[],"
+            + "\"delegateAdmin\":false}],\"denyPolicyItems\":[],\"allowExceptions\":[],\"denyExceptions\":[],"
+            + "\"dataMaskPolicyItems\":[],\"rowFilterPolicyItems\":[],\"id\":40,\"guid\":"
+            + "\"4e2b3406-7b9a-4004-8cdf-7a239c8e2cae\",\"isEnabled\":true,\"version\":1}]}";
```
[the remark this answers is not in the document]
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestRangerLoadTask.java]: Reverted.

File: ql/src/test/org/apache/hadoop/hive/ql/parse/repl/TestCopyUtils.java [hunk 18]
```diff
@@ -97,151 +96,156 @@ public void shouldThrowExceptionOnDistcpFailure() throws Exception {
...
-    }
-    Mockito.verify(fs, Mockito.times(1)).mkdirs(source);
-    try {
-      copyUtils.rename(fs, source, destination);
-    } catch (Exception e) {
-      assertEquals(exception.getClass(), e.getCause().getClass());
+    try (
+            MockedStatic<UserGroupInformation> userGroupInformationMockedStatic = mockStatic(UserGroupInformation.class);
+            MockedStatic<ReplChangeManager> replChangeManagerMockedStatic = mockStatic(ReplChangeManager.class)
+    ) {
```
[the remark this answers is not in the document]
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/parse/repl/TestCopyUtils.java]: Modified.

File: ql/src/test/org/apache/hadoop/hive/ql/parse/repl/TestCopyUtils.java [hunk 19]
```diff
@@ -97,151 +96,156 @@ public void shouldThrowExceptionOnDistcpFailure() throws Exception {
...
-    Mockito.verify(fs, Mockito.times(2)).mkdirs(source);
-    assertEquals (true, copyUtils.rename(fs, source, destination));
-    Mockito.verify(fs, Mockito.times(2)).rename(source, destination);
-    assertEquals (cs, copyUtilsSpy.getContentSummary(fs, source));
-    Mockito.verify(fs, Mockito.times(2)).getContentSummary(source);
-    assertEquals ("dummy", copyUtilsSpy.checkSumFor(source, fs));
+    try (
+            MockedStatic<UserGroupInformation> userGroupInformationMockedStatic = mockStatic(UserGroupInformation.class);
+            MockedStatic<ReplChangeManager> replChangeManagerMockedStatic = mockStatic(ReplChangeManager.class)
+            ) {
```
[the remark this answers is not in the document]
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/parse/repl/TestCopyUtils.java]: Modified.

File: ql/src/test/org/apache/hadoop/hive/ql/testutil/ReflectionUtil.java (hunk 9 above)
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/testutil/ReflectionUtil.java]: Refactored.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestRangerDumpTask.java (hunk 14 above)
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestRangerDumpTask.java]: Reverted.

File: ql/src/test/org/apache/hadoop/hive/ql/parse/repl/load/message/TestPrimaryToReplicaResourceFunction.java (hunk 15 above)
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/parse/repl/load/message/TestPrimaryToReplicaResourceFunction.java]: Reverted.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestAtlasLoadTask.java (hunk 13 above)
Comment (InvisibleProgrammer) [reply ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestAtlasLoadTask.java]: Reverted.

File: ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/message/CreateFunctionHandler.java [hunk 20]
```diff
@@ -155,7 +156,7 @@ static class PrimaryToReplicaResourceFunction
     private String destinationDbName;
 
     PrimaryToReplicaResourceFunction(Context context, MetaData metadata,
-        String destinationDbName) {
+                                     String destinationDbName) {
```
[the remark this answers is not in the document]
Comment (InvisibleProgrammer) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/message/CreateFunctionHandler.java]: Reverted.

**Review by SourabhBadhya — APPROVED** [review]
